lib/spack/spack/compilers/__init__.py

```diff
@@ -125,6 +125,105 @@ def get_compiler_config(scope=None, init_config=True):
     return config
 
 
+def get_compiler_config_from_packages(scope=None):
+    """Return the compiler configuration from packages.yaml"""
+    packages = []
+    config = spack.config.get("packages", scope=scope)
+
+    if not config:
+        return packages
+
+    for name, entry in config.items():
+        if name not in supported_compilers() and name not in package_name_to_compiler_name.keys():
+            continue
+        packages.extend(_compiler_config_from_package_config(entry))
+
+    return packages
+
+
+def _compiler_config_from_package_config(config):
+    externals = config.get("externals", None)
+    if not externals:
+        return []
+
+    compilers = []
+    for entry in externals:
+        compiler = _compiler_config_from_external(entry)
+        if compiler:
+            compilers.append(compiler)
+
+    return compilers
+
+
+def _compiler_config_from_external(config):
+    spec = spack.spec.Spec(config["spec"])
```

Comment:
This should fix the bug:
```suggestion
    spec = spack.spec.parse_with_version_concrete(config["spec"])
```

Reply:
I just finished trying to work around this without being able to find this function and it was driving me crazy, and now I see this review. Thanks!